core/src/main/java/org/apache/carbondata/core/datamap/TableDataMap.java

```diff
@@ -254,6 +251,16 @@ public CarbonTable getTable() {
       // this should not happen
       throw new RuntimeException(" not all the files processed ");
     }
+    if (datamapListForEachThread.size() < numOfThreadsForPruning) {
+      // If the total datamaps less than 2 * numOfThreadsForPruning,
```

Comment:
Can you change the comment to something like "If the number of groups identified are 3 and the numberOfThread for pruning are 4 then the 4th thread will not get any datamaps to prune, therefore set the 'numOfThreadsForPruning' to 3"

Reply:
updated the comment